front/components/navigation/NavigationSidebar.tsx

```diff
@@ -53,13 +62,14 @@ export const NavigationSidebar = React.forwardRef<
         return { ...n, current };
       });
 
-      // Only update navs if the current tab actually changes to prevent blinking effect.
       const isSameCurrent =
         prevNavs.length === newNavs.length &&
         prevNavs.every((prevNav, index) => {
           return prevNav.current === newNavs[index].current;
         });
 
+      const activeTab = nav.filter((n) => n.isCurrent(activePath))[0];
```

Comment:
IIRc some issue when updating the state too often here, it was blinking in the navigation bar.